pkg/api/graph/deployment.go

```diff
@@ -21,8 +21,9 @@ type DeploymentPipelineMap map[*DeploymentConfigNode][]ImagePipeline
 // ImagePipeline represents a build, its output, and any inputs. The input
 // to a build may be another ImagePipeline.
 type ImagePipeline struct {
-	Image ImageTagLocation
-	Build *BuildConfigNode
+	Image       ImageTagLocation
+	ImageStream *ImageStreamNode
```

Comment:
I'd really prefer this to be under ImageTagLocation, such that you can ask the imagetaglocation if it is resolved as being valid.